tests/spread/general/version-git/task.yaml

```diff
@@ -0,0 +1,40 @@
+summary: Use version git
+
+prepare: |
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+
+  mkdir test-snap
+  cd test-snap
+  snapcraft init
+
+  sed -i -e 's/version:.*/version: git/' snap/snapcraft.yaml
+  set_base snap/snapcraft.yaml
+
+  # create an annotated tag
+  git init-db
+  git add .
+  git commit -m msg
+  git tag -a 5.5 -m msg
+  echo >> snap/snapcraft.yaml
+
+restore: |
+  rm -rf test-snap
+  rm -rf ./*.snap
+
+execute: |
+  # Unset SNAPCRAFT_BUILD_ENVIRONMENT=host.
+  unset SNAPCRAFT_BUILD_ENVIRONMENT
+
+  cd test-snap
+
+  # First with lxd
+  snapcraft
```

Comment:
since you are running on all systems, you need to pass --use-lxd here as the default on older snapcrafts is multipass